ngx-tools/src/utilities/object-deep-parse.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Helper function to parse an object with deep keys
+ *
+ * @param obj - the object
+ * @param path - the object path
+ * @return string value at the lowest layer
+ */
+
+export function objectDeepParse(object: { [key: string]: any } | string, keys: string | string[], defaultVal = object): any {
+  if (typeof object === 'string' || typeof object === 'boolean' || !object) {
+    return object;
+  }
+
+  keys = Array.isArray(keys) ? keys : keys.split('.');
```

Comment:
We have a function for this: https://github.com/GetTerminus/ngx-tools/blob/master/ngx-tools/src/README.md#isarray